app/models/concerns/scheduled_event_service.rb

```diff
@@ -0,0 +1,49 @@
+module ScheduledEventService
+  extend ActiveSupport::Concern
+
+  included do
+    # has_many employers
+    def self.individual_market_monthly_enrollment_due_on
+      @individual_market_monthly_enrollment_due_on ||= ScheduledEvent.day_of_month_for("ivl_monthly_enrollment_due_on") || Setting.individual_market_monthly_enrollment_due_on
+    end
+
+    def self.shop_market_binder_payment_due_on
+      @shop_market_binder_payment_due_on ||= ScheduledEvent.day_of_month_for("shop_binder_payment_due_on") || Settings.aca.shop_market.binder_payment_due_on
+    end
+
+    def self.shop_market_initial_application_publish_due_day_of_month
+      @shop_market_initial_application_publish_due_day_of_month ||=
+          ScheduledEvent.where(:event_name => "shop_initial_application_publish_due_day_of_month").first.try(:start_time).try(:day) || Settings.aca.shop_market.initial_application.publish_due_day_of_month
```

Comment:
@nks2109 can you please refer to the method I added to clean up these .where().try().try() calls ? 

I did the first two, please update the remainder